app/src/main/java/com/zulip/android/models/Stream.java

```diff
@@ -120,9 +120,8 @@ public static Stream getByName(ZulipApp app, String name) {
             if (stream == null) {
                 Log.w("Stream.getByName",
                         "We received a stream message for a stream we don't have data for. Fake it until you make it.");
-                stream = new Stream(name);
-                app.getDao(Stream.class).createIfNotExists(stream);
-            }
+                // catch the new stream object with auto-generated id
+                stream = app.getDao(Stream.class).createIfNotExists(new Stream(name));            }
```

Comment:
I should explain more.  I think your code is probably correct for how the Android app works today.  But what I think is going on here that is probably not ideal is that Zulip servers give streams an ID, and the Android app gives streams an ID (here), and those IDs have nothing to do with each other.  :(.